pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/Metadata.java

```diff
@@ -67,36 +68,43 @@ public String toString() {
     }
 
     /**
-     * Class representing item field - name and type.
+     * Class representing item field - name, type, source type, modifiers.
+     * Type - exposed type of field
+     * Source type - type of field in underlying source
+     * Modifiers - additional attributes which describe type or field
      */
     public static class Field {
         private String name;
-        private String type; // TODO: change to enum
+        private EnumHawqType type; // field type which PXF exposes
+        private String sourceType; // filed type PXF reads from
         private String[] modifiers; // type modifiers, optional field
 
-        public Field(String name, String type) {
-
-            if (StringUtils.isBlank(name) || StringUtils.isBlank(type)) {
-                throw new IllegalArgumentException("Field name and type cannot be empty");
-            }
-
-            this.name = name;
-            this.type = type;
+    public Field(String name, EnumHawqType type, String sourceType) {
+        if (StringUtils.isBlank(name) || StringUtils.isBlank(type.getTypeName())
```

Comment:
type could be null, right? maybe just check that it's not null - it probably can't be an empty string because it's an enum.